### DISCUSSION TITLE
Plugin keeps "Loading"

### DISCUSSION THREAD

**Discussion by alexander-hagen** [grafana/grafana-csv-datasource#87] [Q&A] [answered]
I have loaded the CSV plugin in Grafana, running on a Raspberry Pi K3S cluster and it is available in the plugins folder.

When I go to the plug-ins page in Grafana and click on the CSV plugin, the screen says "loading", and it never finishes loading. Any suggestions on what might be wrong, or how best to troubleshoot this issue?

**Comment by marcusolsson** [comment]
I'm not sure why you're experiencing this, but I'm pretty certain it's not related to the plugin itself. think this could be due to your Grafana configuration. Could you check the logs in the browser Developer Tools?

**Comment by alexander-hagen** [comment]
This is what is says:

Fetch error: 404 Not Found Instantiating http://grafana.internal/public/plugins/marcusolsson-csv-datasource/module.js Loading plugins/marcusolsson-csv-datasource/module.

The module is located at /var/lib/grafana/plugins/marcusolsson-csv-datasource., whereas it seems to be looking at usr/share/grafana/public. There is no plugins folder at usr/share/grafana/public. All installed plugins seem to be in public/app/plugins. This may also explain why the csv logo is not displayed as it cannot find the path: /public/plugins/marcusolsson-csv-datasource/img/logo.svg

It seems to be using different folders. My version of Grafana is v8.0.3, in case that matters.

**Reply by marcusolsson** [reply]
Could you see if the answer to [this question](https://github.com/marcusolsson/grafana-json-datasource/discussions/134) works for you as well?

**Comment by alexander-hagen** [comment, marked as answer]
I managed to fix this. The root cause was that the plugins folder resides on a NFS share which was vFAT. That allows setting permissions at mount point only, and fails at all other levels. I have changed the NFS share to ext4, and now all is well.
